tests/libnet/cloudinit.go

```diff
@@ -23,12 +23,106 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+func NewNetworkData(options ...NetworkDataOption) (string, error) {
+	networkData := CloudInitNetworkData{
+		Version: 2,
+	}
+
+	for _, option := range options {
+		err := option(&networkData)
+		if err != nil {
+			return "", nil
+		}
+	}
+
+	nd, err := yaml.Marshal(&networkData)
+	if err != nil {
+		return "", err
+	}
+
+	return string(nd), nil
+}
+
+func WithEthernet(name string, options ...NetworkDataInterfaceOption) NetworkDataOption {
+	return func(networkData *CloudInitNetworkData) error {
+		if networkData.Ethernets == nil {
+			networkData.Ethernets = map[string]CloudInitInterface{}
+		}
+
+		networkDataInterface := CloudInitInterface{name: name}
+
+		for _, option := range options {
+			err := option(&networkDataInterface)
+			if err != nil {
+				return nil
+			}
+		}
+
+		networkData.Ethernets[name] = networkDataInterface
+		return nil
+	}
+}
+
+func WithAddresses(addresses ...string) NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		if networkDataInterface.Addresses == nil {
+			networkDataInterface.Addresses = []string{}
+		}
+		networkDataInterface.Addresses = append(networkDataInterface.Addresses, addresses...)
+		return nil
+	}
+}
+
+func WithDHCP4() NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		enabled := true
+		networkDataInterface.DHCP4 = &enabled
+		return nil
+	}
+}
+
+func WithStaticIPv6Address() NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		WithAddresses(DefaultIPv6Address)(networkDataInterface)
+		networkDataInterface.Gateway6 = DefaultIPv6Gateway
+		return nil
+	}
+}
+
+func WithNameserverFromCluster() NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		dnsServerIP, err := ClusterDNSServiceIP()
+		if err != nil {
+			return err
+		}
+		networkDataInterface.Nameservers = CloudInitNameservers{
+			Addresses: []string{dnsServerIP},
+			Search:    SearchDomains(),
+		}
+		return nil
+	}
+}
+
+func MatchingMAC(macAddress string) NetworkDataInterfaceOption {
+	return func(networkDataInterface *CloudInitInterface) error {
+		networkDataInterface.Match = CloudInitMatch{
+			MACAddress: macAddress,
+		}
+		networkDataInterface.SetName = networkDataInterface.name
+		return nil
+	}
+}
+
+type NetworkDataOption func(*CloudInitNetworkData) error
+type NetworkDataInterfaceOption func(*CloudInitInterface) error
```

Comment:
I think these fit at the head of the file.

Reply:
Done